Source/WebCore/css/ComputedStyleExtractor.cpp

```diff
@@ -2304,14 +2304,19 @@ static bool rendererCanHaveTrimmedMargin(const RenderBox& renderer, std::optiona
 
     // containingBlock->isBlockContainer() can return true even if the item is in a RenderFlexibleBox
     // (e.g. buttons) so we should explicitly check that the item is not a flex item to catch block containers here
-    if (!renderer.isFlexItem() && (containingBlock->isRenderGrid() || containingBlock->isBlockContainer()))
+    if (!renderer.isFlexItem() && containingBlock->isBlockContainer())
         return false;
 
     if (containingBlock->isFlexibleBox()) {
         if (!marginTrimType)
             return containingBlock->style().marginTrim().containsAny({ MarginTrimType::BlockStart, MarginTrimType::InlineEnd });
         return containingBlock->style().marginTrim().contains(marginTrimType.value());
     }
+    if (containingBlock->isRenderGrid()) {
+        if (!marginTrimType)
+            return containingBlock->style().marginTrim().containsAny( { MarginTrimType::BlockStart });
```

Comment:
In other places you are using `.contains`. I am not sure if `containsAny` is getting us anything since there is only one element.

Reply:
Yeah that's fair since in this patch block-start is the only valid value to consider in this context. Went ahead and changed it